aten/src/ATen/native/GridSampler.cpp

```diff
@@ -13,8 +16,23 @@ using at::native::detail::GridSamplerInterpolation;
 using at::native::detail::GridSamplerPadding;
 
 namespace {
-  static inline int64_t clip_coordinates(int64_t in, int64_t clip_limit) {
-    return std::min(clip_limit - 1, std::max(in, static_cast<int64_t>(0)));
+  template<typename scalar_t>
+  static inline scalar_t clip_coordinates(scalar_t in, int64_t clip_limit) {
```

Comment:
I'm confused: why did `in` turn from an `int64_t` index to a `scalar_t`? Isn't it a coordinate?

Reply:
@ezyang I updated the PR description with a more detailed explanation on what changed. :) This should be ready for review now.